cirq-web/cirq_ts/src/bloch_sphere/components/meridians_test.ts

```diff
@@ -0,0 +1,189 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {expect} from 'chai';
+import {Meridians} from './meridians';
+import {Orientation} from './enums';
+import {Vector3} from 'three';
+
+describe('Meridians', () => {
+  const DEFAULT_RADIUS = 5;
+  const DEFAULT_H_MERIDIANS = 7;
+  const DEFAULT_V_MERIDIANS = 4;
+
+  describe('defaults', () => {
+    it('createHorizontalChordMeridians() generates the correct number of lines given the default number (7)', () => {
+      const meridians = new Meridians(
+        DEFAULT_RADIUS,
+        DEFAULT_H_MERIDIANS,
+        Orientation.HORIZONTAL_CHORD
+      );
+      expect(meridians.children.length).to.equal(7);
+    });
+
+    it('createHorizontalCircleMeridians() generates the correct number of lines given the default number (7)', () => {
+      const meridians = new Meridians(
+        DEFAULT_RADIUS,
+        DEFAULT_H_MERIDIANS,
+        Orientation.HORIZONTAL
+      );
+      expect(meridians.children.length).to.equal(7);
+    });
+
+    it('createVerticalMeridians() generates the correct number of lines given the default number (4)', () => {
+      const meridians = new Meridians(
+        DEFAULT_RADIUS,
+        DEFAULT_V_MERIDIANS,
+        Orientation.VERTICAL
+      );
+      expect(meridians.children.length).to.equal(4);
+    });
+
+    it('createHorizontalChordMeridians() generates lines at the correct positions with defaults', () => {
+      const meridians = new Meridians(
+        DEFAULT_RADIUS,
+        DEFAULT_H_MERIDIANS,
+        Orientation.HORIZONTAL_CHORD
+      );
+
+      const positions = [
+        new Vector3(0, 0, 0),
+        new Vector3(0, 4.5, 0),
+        new Vector3(0, -4.5, 0),
+        new Vector3(0, 3, 0),
+        new Vector3(0, -3, 0),
+        new Vector3(0, 1.5, 0),
+        new Vector3(0, -1.5, 0),
+      ];
+
+      meridians.children.forEach((el, index) => {
+        expect(el.position).to.eql(positions[index]);
+      });
+    });
+  });
+
+  describe('configurables', () => {
+    it('createHorizontalChordMeridians() generates the correct number of lines given valid numbers', () => {
```

Comment:
I found a bug :) I tested with 1 (playing around with the visualization provided by Suresh) and it creates 3 horizontal lines - this test fails for 1 `lineValue`, I'd expect 1 `lineNumber`, but it gives 3. 

Also, for 4 why do we expect 5?

Same thing visually: 

![image](https://user-images.githubusercontent.com/635073/123001861-36e1b100-d37f-11eb-875b-a0951cf870b2.png)

Reply:
I outline this in the docstring for building Meridians (https://github.com/seunomonije/Cirq/blob/d59d56aa67cdd2bf7ff922d2c91dc83938bd2937/cirq-web/cirq_ts/src/bloch_sphere/components/meridians.ts#L89). 

Honestly it was a personal asethetic choice, I like a minimum of three for the horizontal chord meridians. I can add cases for 1 and 2 though if you feel strongly about it.

And in the same comments, an odd number of horizontal meridians is needed because I think asethetically there needs to be an equator meridian. Then all the other meridians can go based off of that..if the number is even, then the meridians are unbalanced.

Reply:
Ah, got it, that makes more sense. What you just described should be documented in the test cases and a comment would help. The tests serve as a living documentation (besides the doc string which is good for publishing stuff off of source code).

I think I do want to experiment with a single meridian given Suresh's design: 

<img src="https://user-images.githubusercontent.com/635073/123004455-9d1c0300-d382-11eb-8a09-82b01a477734.png" height='200px'>

Reply:
Bet..i'll add support for single horizontal merdians. That design looks cool too.